openpype/hosts/maya/plugins/publish/extract_import_reference.py

```diff
@@ -0,0 +1,124 @@
+import os
+
+from maya import cmds
+
+import pyblish.api
+
+from openpype.pipeline import publish, legacy_io
+from openpype.settings import get_project_settings
+from openpype.hosts.maya.api import lib
+
+
+def _get_project_setting():
+    project_name = legacy_io.active_project()
+    project_setting = get_project_settings(project_name)
+    maya_enabled = (
+        project_setting["maya"]["publish"]["ImportReference"]["enabled"]
+    )
+    use_published = (
+        project_setting["deadline"]["publish"]["MayaSubmitDeadline"]["use_published"] # noqa
+    )
+    if maya_enabled != use_published:
+        return False
+    else:
+        return use_published
+
+
+class ImportReference(publish.Extractor):
+    """
+
+        Extract the scene with imported reference.
+        The temp scene with imported reference is
+        published for rendering if this extractor is activated
+
+    """
+
+    label = "Import Reference"
+    order = pyblish.api.ExtractorOrder - 0.48
+    hosts = ["maya"]
+    families = ["renderlayer", "workfile"]
+    active = _get_project_setting()
```

Comment:
Setting the active state like this probably won't work because it'd get overridden by OpenPype settings for plug-in discovery due to the `project_settings/maya/publish/ImportReference/enabled` setting existence.

I suspect this is wrong.

Also, it feels off that we're toggling this based on `MayaSubmitDeadline/use_published` setting but it also applies to `workfile` family - which means that it'd ALWAYS trigger for every workfile, even if not used for rendering.

We're mixing logic here.

Reply:
I am still thinking about which families is the best suited for this case, as it is asking the system to publish the temporary scene as a published workfile, while saving a temporary scene as a "current" workfile to use for rendering if the published workfile option is turned off. Perhaps we can introduce `optional = True ` in this extractor so that the user can choose to turn on/off in the publisher